### ISSUE TITLE
Page modification request: find

### ISSUE THREAD

**Issue by zmajeed** [tldr-pages/tldr#15107]
### Command description

Two `find` examples are wrong or misleading

1. The use of globstar `**` below is misleading

- Find files matching multiple path/name patterns:
`find {{root_path}} -path '{{**/path/**/*.ext}}' -or -name '{{*pattern*}}'`

`find` does not support globstar - it uses `fnmatch()` for glob patterns which doesn't recognize globstar syntax - `find` does not set the `FNM_PATHNAME` flag for `fnmatch()` - this means a single star `*` matches anything including slash in paths and two consecutive stars are effectively  redundant

The example should be

- Find files matching multiple path/name patterns:
`find {{root_path}} -path '{{*/path/*/*.ext}}' -or -name '{{*pattern*}}'`

2. The following example has wrong syntax - it uses pipe `|` instead of comma `,` to match multiple types - also pipe is not escaped for the shell which interprets the command as a pipeline and would be an error or cause unexpected results

- Find empty files (0 byte) or directories and delete them verbosely:
`find {{root_path}} -type {{f|d}} -empty -delete -print`

From https://manned.org/man/find

```
-type c
              File is of type c:

              To search for more than one type at once, you can supply the
              combined list of type letters separated by a comma `,' (GNU
              extension).
```

The example should be

- Find empty files (0 byte) or directories and delete them verbosely:
`find {{root_path}} -type '{{f,d}}' -empty -delete -print`

### Documentation

https://manned.org/man/find

### Platform

Linux

### VCS repository link (e.g. GitHub, GitLab)

https://git.savannah.gnu.org/cgit/findutils.git

**Comment by gutjuri** [comment]
> The following example has wrong syntax - it uses pipe | instead of comma , to match multiple types - also pipe is not escaped for the shell which interprets the command as a pipeline and would be an error or cause unexpected results

The `|` is tldr-syntax (see styleguide).

**Comment by zmajeed** [comment]
Not sure I follow ... Are you saying the `find` commandline example is not meant to be run as-is?

**Comment by zmajeed** [comment]
If the pipe is meant to say `-type` only takes `f` or `d` then that's misleading because it can take a few other letters - and with the comma `,` separator it can take multiple letters at the same time

**Comment by gutjuri** [comment]
> Not sure I follow ... Are you saying the `find` commandline example is not meant to be run as-is?

No, the examples are generally meant to be modified before running. E.g. we have this example https://github.com/tldr-pages/tldr/blob/9a51987973bdcc91026ba89452f33b6005d7f7c1/pages/common/7z.md?plain=1#L28 which includes some possible options that can be passed to 7zip. 

However, I think that you are right that in this particular case, the example is misleading, especially with the wording of the description. So yes, we should probably change it.

**Comment by moxtong2** [comment]
hei，brother，How to set up tldr tar to output Chinese comments？

**Comment by kbdharun** [comment]
> hei，brother，How to set up tldr tar to output Chinese comments？

Hi, in future feel free to open a [new discussion](https://github.com/tldr-pages/tldr/discussions) or issue for your general queries since it's offtopic to the current page modification request.

---

On your installed tldr client, you can pass the language (`-L`) flag and specify your language code to view the translated page (if it exists), in this case, if you type something like `tldr -L zh tar` then you will get the [Chinese translation of the page](https://github.com/tldr-pages/tldr/blob/main/pages.zh/common/tar.md). Most of our clients support reading the system locale setting so if your system language is set to Chinese then you will be displayed the respective tldr page in it, in case it's still in English refer to your client's documentation to set an environment variable to set the language permanently.

**Comment by acuteenvy** [comment]
Changes from #15108 have been implemented in #15542.
